### PR TITLE
Refactor page permission actions

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Refactored the `page_actions.py`

### Proposed changes
- Move page permission actions to separate file
- Introduce `AbstractPagePermission` class, to implement template method pattern to avoid code duplication 
- Split up big functions in smaller functions to increase readability of the code

### Background:
I did some code analysis of our code (integreat-app, integreat-cms and entitlementcard). One result was that the page_actions.py is both a hotspot (=changed frequently) and has one of the lowest code health in our whole codebase. Therefore I created this PR to improve the code quality of this file.

I will show some insights about the code analysis I made in a barcamp at the conference.

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by MizukiTemma — APPROVED** [review]
Cool improvement 😸

**Review by JoeyStk — CHANGES_REQUESTED** [review]
Thank you very much for your efforts! This is a great contribution!
I only have a few minor (and admittedly nit-picky) comments :)

**Review by JoeyStk — APPROVED** [review]

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -8931,6 +8880,57 @@ msgstr ""
...
+msgid ""
+"Page-specific permissions cannot be granted to users who do not have "
+"permission to view pages (e.g event managers)."
+msgstr ""
+"Seitenspezifische Berechtigungen können nicht an Benutzer:innen vergeben "
+"werden, die keine Seiten sehen dürfen (z.B. Terminplaner:innen)."
+
+#: cms/views/pages/page_permission_actions.py
+msgid "Information: The user \"{}\" has this permission already."
+msgstr "Information: \"{}\" hat diese Berechtigung bereits"
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply integreat_cms/locale/de/LC_MESSAGES/django.po]: You are right, added the dot.

File: integreat_cms/cms/views/pages/page_permission_actions.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    def save_revoke_permission(self, user: User, page: Page) -> None:
+        if user in page.editors.all():
+            page.editors.remove(user)
+            page.save()
+
+
+def __render_response(
+    request: HttpRequest, page: Page, permission_message: PermissionMessage
+) -> HttpResponse:
+    print(permission_message.level_tag.value)
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply integreat_cms/cms/views/pages/page_permission_actions.py]: removed it.

**Pull request merged**